### PR TITLE
Cleans up internal temp code

### PR DESCRIPTION
**Pull request by ghost**
ripped straight from spacepods... also no changelog due this being 100% code-side change, no noticeable in game effects.

### CODE REVIEW

**Comment by kevinz000** [comment]
I SERIOUSLY HOPE YOU TESTED THIS.
BECAUSE YOU KNOW IF YOU FUCK UP ATMOS CODE IT'LL BE BAD.

**Comment by MoreRobustThanYou** [comment]
yeah this was tested during much of spacepod testing

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 1]
```diff
@@ -393,4 +393,11 @@
 /obj/machinery/atmospherics/components/unary/cryo_cell/can_see_pipes()
 	return 0 //you can't see the pipe network when inside a cryo cell.
 
+/obj/machinery/atmospherics/components/unary/cryo_cell/return_temperature()
+	var/datum/gas_mixture/G = AIR1
```
Comment (nicbn) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: Why aren't you using AIR1 instead of G?

**Comment by nicbn** [comment]
Also adding lines of code is the inverse of cleanup.

**Comment by MoreRobustThanYou** [comment]
@nicbn this is removing hardcoded checks

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm (hunk 1 above)
Comment (MoreRobustThanYou) [reply code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: this was copy pasted from the original hardcoded cryo return temp code

**Comment by MoreRobustThanYou** [comment]
@tgstation/commit-access please review and/or merge

**Comment by optimumtact** [comment]
conflicting but should be good for merge after

**Comment by MoreRobustThanYou** [comment]
fixed

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm (hunk 1 above)
Comment (optimumtact) [reply code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: it's supposed to be slightly faster to get a local var reference to the global

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -707,28 +707,18 @@
...
-			loc_temp = environment.temperature
-		else
-			loc_temp = G.temperature
-
 	else
 		loc_temp = environment.temperature
+	return loc_temp
+
 
 	return loc_temp
```
Comment (Cyberboss) [inline code/modules/mob/living/living.dm]: You doubled up

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/objs.dm [hunk 3]
```diff
@@ -226,3 +226,6 @@
 		current_skin = choice
 		icon_state = unique_reskin[choice]
 		to_chat(M, "[src] is now skinned as '[choice].'")
+
+/obj/proc/return_temperature()
+	return
```
Comment (Cyberboss) [inline code/game/objects/objs.dm]: Make it an atom proc please

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 4]
```diff
@@ -450,3 +450,7 @@
 
 /obj/structure/closet/AllowDrop()
 	return TRUE
+
+
+/obj/structure/closet/return_temperature()
+	return
```
Comment (Cyberboss) [inline code/game/objects/structures/crates_lockers/closets.dm]: Why does this need to exist?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/crates_lockers/closets.dm (hunk 4 above)
Comment (MoreRobustThanYou) [reply code/game/objects/structures/crates_lockers/closets.dm]: there was a bug where you'd be temperature proof inside lockers IIRC

File: code/game/objects/objs.dm (hunk 3 above)
Comment (MoreRobustThanYou) [reply code/game/objects/objs.dm]: done

File: code/modules/mob/living/living.dm [hunk 5]
```diff
@@ -707,29 +707,16 @@
...
-	if(istype(loc, /obj/mecha))
-		var/obj/mecha/M = loc
-		loc_temp =  M.return_temperature()
-
-	else if(istype(loc, /obj/structure/transit_tube_pod))
-		loc_temp = environment.temperature
-
+	if(isobj(loc))
+		var/obj/oloc = loc
+		var/obj_temp = oloc.return_temperature()
```
Comment (AnturK) [inline code/modules/mob/living/living.dm]: You just made anything that does not have this defined, like closets use T0C instead of environment.

**Pull request merged**